packages/sdk/src/types/CommonTypes.ts

```diff
@@ -411,15 +419,59 @@ export interface ActionEventErrorData {
 }
 
 interface AsyncErrorBase {
+    type: AsyncErrorType;
     message: string;
 }
 
-export interface ActionAsyncError extends AsyncErrorBase {
+export class ActionAsyncError implements AsyncErrorBase {
+    type: AsyncErrorType;
     id?: string;
     event?: ActionEditorEvent;
     eventChain?: ActionEventErrorData[];
+    message: string;
+
+    constructor(message: string, id?: string, event?: ActionEditorEvent, eventChain?: ActionEventErrorData[]) {
+        this.type = AsyncErrorType.action;
+        this.message = message;
+        this.id = id;
+        this.event = event;
+        this.eventChain = eventChain;
+    }
+}
+
+export class DataRowAsyncError implements AsyncErrorBase {
+    type: AsyncErrorType;
+    count: number;
+    exceptions: EditorExceptionDto[];
+    message: string;
+
+    constructor(count: number, message: string, exceptions: EditorExceptionDto[]) {
+        this.type = AsyncErrorType.dataRow;
+        this.count = count;
+        this.message = message;
+        this.exceptions = exceptions;
+    }
+}
+
+export enum AsyncErrorType {
+    action = 'action',
+    dataRow = 'dataRow',
+}
+
+export interface ExceptionContext {
+    variableId?: Id;
+    variableType?: VariableType;
+    variableName?: string;
+    variableLabel?: string;
+}
```

Comment:
The only mandatory one is `variableId` IMO, what does the rest of the team think?

```suggestion
export interface ExceptionContext {
    variableId: Id;
}
```

Reply:
@psamusev @alexandraFlavia9

Reply:
I don't like that _"if in the future something needs more info regarding the variable, the engine will need to make a change to include this info as well."_. 

@alexandraFlavia9 would it be an actual SDK call or do you have the variables stored somewhere locally?